src/nnbench/runner.py

```diff
@@ -247,6 +249,12 @@ def run(
         if not self.benchmarks:
             self.collect(path_or_module, tags)
 
+        family_sizes: dict[str, Any] = collections.defaultdict(int)
+        for bm in self.benchmarks:
+            family_sizes[bm.fn.__name__] += 1
+
+        family_indices: dict[str, Any] = collections.defaultdict(int)
+
```

Comment:
I would put this below the `if not self.benchmarks` check in L266, because we don't need to run it if there are no benchmarks.